src/Network/Datadog/Event.hs

```diff
@@ -0,0 +1,271 @@
+  toJSON Error = Data.Aeson.String "error"
+  toJSON Warning = Data.Aeson.String "warning"
+  toJSON Info = Data.Aeson.String "info"
+  toJSON Success = Data.Aeson.String "success"
+
+instance FromJSON AlertType where
+  parseJSON (Data.Aeson.String "error") = return Error
+  parseJSON (Data.Aeson.String "warning") = return Warning
+  parseJSON (Data.Aeson.String "info") = return Info
+  parseJSON (Data.Aeson.String "success") = return Success
+  parseJSON (Data.Aeson.String s) = fail $ "AlertType: String \"" ++ unpack s ++ "\" is not a valid AlertType"
+  parseJSON a = modifyFailure ("AlertType: " ++) $ typeMismatch "String" a
+
+
+-- | A source from which an event may originate, recognized by Datadog.
+data SourceType = Nagios
+                | Hudson
+                | Jenkins
+                | User
+                | MyApps
+                | Feed
+                | Chef
+                | Puppet
+                | Git
+                | BitBucket
+                | Fabric
+                | Capistrano
+                deriving (Eq)
+
+instance Show SourceType where
+  show Nagios = "nagios"
+  show Hudson = "hudson"
+  show Jenkins = "jenkins"
+  show User = "user"
+  show MyApps = "my apps"
+  show Feed = "feed"
+  show Chef = "chef"
+  show Puppet = "puppet"
+  show Git = "git"
+  show BitBucket = "bitbucket"
+  show Fabric = "fabric"
+  show Capistrano = "capistrano"
+
+instance ToJSON SourceType where
+  toJSON Nagios = Data.Aeson.String "nagios"
+  toJSON Hudson = Data.Aeson.String "hudson"
+  toJSON Jenkins = Data.Aeson.String "jenkins"
+  toJSON User = Data.Aeson.String "user"
+  toJSON MyApps = Data.Aeson.String "my apps"
+  toJSON Feed = Data.Aeson.String "feed"
+  toJSON Chef = Data.Aeson.String "chef"
+  toJSON Puppet = Data.Aeson.String "puppet"
+  toJSON Git = Data.Aeson.String "git"
+  toJSON BitBucket = Data.Aeson.String "bitbucket"
+  toJSON Fabric = Data.Aeson.String "fabric"
+  toJSON Capistrano = Data.Aeson.String "capistrano"
+
+instance FromJSON SourceType where
+  parseJSON (Data.Aeson.String "nagios") = return Nagios
+  parseJSON (Data.Aeson.String "hudson") = return Hudson
+  parseJSON (Data.Aeson.String "jenkins") = return Jenkins
+  parseJSON (Data.Aeson.String "user") = return User
+  parseJSON (Data.Aeson.String "my apps") = return MyApps
+  parseJSON (Data.Aeson.String "feed") = return Feed
+  parseJSON (Data.Aeson.String "chef") = return Chef
+  parseJSON (Data.Aeson.String "puppet") = return Puppet
+  parseJSON (Data.Aeson.String "git") = return Git
+  parseJSON (Data.Aeson.String "bitbucket") = return BitBucket
+  parseJSON (Data.Aeson.String "fabric") = return Fabric
+  parseJSON (Data.Aeson.String "capistrano") = return Capistrano
+  parseJSON (Data.Aeson.String s) = fail $ "SourceType: String \"" ++ unpack s ++ "\" is not a valid SourceType"
+  parseJSON a = modifyFailure ("SourceType: " ++) $ typeMismatch "String" a
+
+
+-- | Details that describe an event.
+data EventSpec = EventSpec { edTitle :: Text
```

Comment:
That's a work in progress.  Getting there.  (I've never actually created lenses before.)